apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.ejs

```diff
@@ -84,32 +84,80 @@
+                  <% sharing_sets_in.forEach(function(sharing_set) { %>
+                    <span class="badge color-gray1"><%= sharing_set.name %></span>
+                  <% }); %>
+                  <% if (authz_data.has_course_permission_edit && !course.example_course) { %>
+                    <% if (sharing_sets_other.length > 0) { %>
+                    <form name="sharing-set-add" method="POST" class="d-inline">
+                      <input type="hidden" name="__action" value="sharing_set_add">
+                      <input type="hidden" name="__csrf_token" value="<%= __csrf_token %>">
+                      <div class="btn-group btn-group-sm" role="group">
+                        <button id="addSharingSet" type="button" class="btn btn-sm btn-outline-dark dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+                          Add...
+                        </button>
+                        <div class="dropdown-menu" aria-labelledby="addSharingSet">
+                            <% sharing_sets_other.forEach(function(sharing_set) { %>
+                              <button class="dropdown-item" type="submit" name="unsafe_sharing_set_id" value="<%= sharing_set.id %>">
+                                <%= sharing_set.name %>
+                              </button>
+                          <% }); %>
+                        </div>
+                      </div>
+                    </form>
+                    <% } %>
+                    <button class="btn btn-sm btn-outline-primary" type="button" data-toggle="modal" data-target="#publiclyShareModal">
+                      Share Publicly
+                    </button>
+                    <div class="modal fade" id="publiclyShareModal" tabindex="-1" role="dialog" aria-labelledby="publiclyShareModalTitle" aria-hidden="true">
+                      <div class="modal-dialog modal-dialog-centered" role="document" >
+                        <div class="modal-content">
+                          <div class="modal-header">
+                            <h5 class="modal-title" id="publiclyShareModalTitle">
+                              Confirm Publicly Share Question
+                            </h5>
+                            <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                              <span aria-hidden="true">&times;</span>
                             </button>
-                        <% }); %>
+                          </div>
+                          <div class="modal-body">
+                            <p>
+                              Are you sure you want to publicly share this question?
+                            </p>
+                            <p>
+                              Once this question is publicly shared, anyone will be able to view it
+                              or use it as a part of their course. This operation cannot be undone.
+                            </p>
```

Comment:
I think we are going to want some legalese here to make sure that the instructor is granting PLInc a license to share their content, and optionally to license it under an open license (e.g., a CC license). I'll work on some text for this.

For similar reasons we should also move this modal into a component in the `ee/` directory. Here's an example of making a modal: https://github.com/PrairieLearn/PrairieLearn/blob/588b91e5a63253ccc67422124e51dd3658210aa3/apps/prairielearn/src/pages/administratorFeatures/administratorFeatures.html.ts#L104 https://github.com/PrairieLearn/PrairieLearn/blob/588b91e5a63253ccc67422124e51dd3658210aa3/apps/prairielearn/src/pages/administratorFeatures/administratorFeatures.html.ts#L207

Reply:
How are people supposed to run their own PL instance if we have a non-EE page using an EE modal?

CC license seems fine when talking about rendered questions being shared but how is this going to change when we start doing source-available as well? Code-oriented licenses like MIT or GPL start to make more sense over some level of questions complexity.

I don't expect you to have all the answer right now, just making sure we have this conversations.

Reply:
@mwest1066 do you have this figured out yet?

Reply:
Let's add the following (please check the links are correct!):
```suggestion
<% if (isEnterprise) { %>
<p>
You retain full ownership of all shared content as described in the <a href="https://www.prairielearn.com/legal/terms#2-user-content">Terms of Service</a>. To allow PrairieLearn to share your content to other users you agree to the <a href="https://www.prairielearn.com/legal/terms#3-user-content-license-grant">User Content License Grant</a>.
</p>
<% } %>
```

Reply:
As discussed in meeting, we are going to wait on worrying about CC licensing issues until later when we add the feature for questions to be source-available.